pkg/otlp/metrics/metrics_translator_test.go

```diff
@@ -371,6 +373,221 @@ func TestMapIntMonotonicWithReboot(t *testing.T) {
+		met := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+		met.SetName(exampleDims.name)
+		met.SetEmptySum()
+		met.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+		met.Sum().SetIsMonotonic(true)
+		dpsInt := met.Sum().DataPoints()
+		dpsInt.EnsureCapacity(3)
+
+		dpInt := dpsInt.AppendEmpty()
+		dpInt.SetStartTimestamp(seconds(startTs))
+		dpInt.SetTimestamp(seconds(startTs + 3))
+		dpInt.SetIntValue(10)
+
+		dpInt2 := dpsInt.AppendEmpty()
+		dpInt2.SetStartTimestamp(seconds(startTs))
+		// lower timestamp than dpInt. This point should be ignored.
+		dpInt2.SetTimestamp(seconds(startTs + 2))
+		dpInt2.SetIntValue(25)
+
+		dpInt3 := dpsInt.AppendEmpty()
+		dpInt3.SetStartTimestamp(seconds(startTs))
+		dpInt3.SetTimestamp(seconds(startTs + 5))
+		dpInt3.SetIntValue(40)
+
+		ctx := context.Background()
+		tr := newTranslator(t, zap.NewNop())
+		consumer := &mockFullConsumer{}
+
+		rmt, _ := tr.MapMetrics(ctx, md, consumer)
+		assert.ElementsMatch(t,
+			consumer.metrics,
+			[]metric{
+				newCountWithHost(exampleDims, uint64(seconds(startTs+3)), 10, fallbackHostname),
+				newCountWithHost(exampleDims, uint64(seconds(startTs+5)), 30, fallbackHostname),
+			},
+		)
+		assert.Empty(t, rmt.Languages)
+	})
+}
+
+// Regression Test: This test validates that a point (the first point in a NumberDataPointSlice) with a timestamp older or equal
+// to the timestamp of previous point received is dropped and not computed as a first val.
+func TestMapIntMonotonicDropPointPointBeginningOfSlice(t *testing.T) {
+	t.Run("equal", func(t *testing.T) {
+		tr := newTranslator(t, zap.NewNop())
+		dims := &Dimensions{name: exampleDims.name, host: fallbackHostname}
+		startTs := int(getProcessStartTime()) + 1
+		md := pmetric.NewMetrics()
+		met := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+		met.SetName(dims.name)
+		met.SetEmptySum()
+		met.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+		met.Sum().SetIsMonotonic(true)
+		dpsInt := met.Sum().DataPoints()
+		dpsInt.EnsureCapacity(3)
+
+		// dpInt1
+		tr.prevPts.MonotonicDiff(dims, uint64(seconds(startTs)), uint64(seconds(startTs+2)), 10)
+
+		dpInt2 := dpsInt.AppendEmpty()
+		dpInt2.SetStartTimestamp(seconds(startTs))
+		// duplicate timestamp to dpInt. This point should be ignored and not used as first val
+		dpInt2.SetTimestamp(seconds(startTs + 2))
+		dpInt2.SetIntValue(20)
+
+		dpInt3 := dpsInt.AppendEmpty()
+		dpInt3.SetStartTimestamp(seconds(startTs))
+		dpInt3.SetTimestamp(seconds(startTs + 4))
+		dpInt3.SetIntValue(40)
+
+		ctx := context.Background()
+		consumer := &mockFullConsumer{}
+
+		rmt, _ := tr.MapMetrics(ctx, md, consumer)
+		assert.ElementsMatch(t,
+			consumer.metrics,
+			[]metric{
+				newCountWithHost(exampleDims, uint64(seconds(startTs+4)), 30, fallbackHostname),
+			},
+		)
+		assert.Empty(t, rmt.Languages)
+	})
+
+	t.Run("older", func(t *testing.T) {
```

Comment:
This test fails on `main` without the changes in this PR. With the old logic, an old point could be computed as a first val erroneously.